Marlin/src/HAL/HAL_LPC1768/HAL.h

```diff
@@ -132,11 +132,40 @@ int freeMemory();
                                     // Memory usage per ADC channel (bytes): 4 (32 Bytes for 8 channels)
 
 using FilteredADC = LPC176x::ADC<ADC_LOWPASS_K_VALUE, ADC_MEDIAN_FILTER_SIZE>;
-#define HAL_adc_init()         FilteredADC::init()
+
+extern uint32_t HAL_adc_reading;
+[[gnu::always_inline]] inline void HAL_start_adc(const pin_t pin) {
+  HAL_adc_reading = FilteredADC::read(pin) >> 6; // returns 16bit value, reduce to 10bit
+}
+[[gnu::always_inline]] inline uint16_t HAL_read_adc() {
```

Comment:
As far as I know `FORCE_INLINE` is available here.

Reply:
Indeed it is, I sometimes forget to switch back to Marlin convention, I tend to avoid the preprocessor by any means necessary.